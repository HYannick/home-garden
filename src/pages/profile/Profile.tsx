import React from 'react';
import { useTranslation } from 'react-i18next';
import { useGetUserInfos } from '../home/home.hooks';
import { useGetPlantsCount } from './Profile.hooks';

import ProfileView from './Profile.view';
import { ChartItemProps, ProfileViewProps } from './Profile.types';

const Profile: React.FC = (props) => {
  const { t } = useTranslation();

  const { loading: userLoading, userInfos } = useGetUserInfos();
  const { loading: countLoading, counters } = useGetPlantsCount();

  const chartData: ChartItemProps[] = [
    { name: 'healthy', label: 'Healthy', variant: 'primary', value: counters.healthy },
    { name: 'warning', label: 'Warning', variant: 'warning', value: counters.warning },
    { name: 'needy', label: 'Needy', variant: 'danger', value: counters.danger },
  ];

  const viewProps: ProfileViewProps = {
    ...props,
    t,
    userLoading,
    countLoading,
    userInfos,
    chartData,
    counters,
  };

  return <ProfileView {...viewProps}/>;
};

export default Profile;
