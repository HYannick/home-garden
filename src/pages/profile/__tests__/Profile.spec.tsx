import React from 'react';
import 'jest-dom/extend-expect';
import { cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { renderWithTheme } from '../../../theme-wrapper';
import ProfileView from '../Profile.view';
import { ProfileViewProps } from '../Profile.types';

afterEach(cleanup);
describe('Profile Page', () => {
  const defaultProps: ProfileViewProps = {
    t: jest.fn().mockImplementation(() => 'profile_page.loading'),
    userLoading: false,
    countLoading: false,
    userInfos: {
      username: 'Mathilde',
      avatar: 'cute-kitten.jpg',
    },
    chartData: [],
    counters: [],
  };

  it('should render loading text properly', () => {
    const props: ProfileViewProps = {
      ...defaultProps,
      userLoading: true,
    };
    const { container } = renderWithTheme(
      <Router><ProfileView {...props} /></Router>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render properly', () => {
    const { container } = renderWithTheme(
      <Router><ProfileView {...defaultProps} /></Router>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should have a loading message when userLoading is true', () => {
    const props: ProfileViewProps = {
      ...defaultProps,
      userLoading: true,
    };
    const { getByText } = renderWithTheme(
      <Router><ProfileView {...props} /></Router>,
    );

    expect(getByText(/^profile_page.loading/)).toHaveTextContent('profile_page.loading');
  });

  it('should have a loading message when countLoading is true', () => {
    const props: ProfileViewProps = {
      ...defaultProps,
      countLoading: true,
    };
    const { getByText } = renderWithTheme(
      <Router><ProfileView {...props} /></Router>,
    );

    expect(getByText(/^profile_page.loading/)).toHaveTextContent('profile_page.loading');
  });

  it('should display the username', () => {
    const { getByTestId } = renderWithTheme(
      <Router><ProfileView {...defaultProps} /></Router>,
    );

    expect(getByTestId('username')).toHaveTextContent('Mathilde');
  });

  it('should display the chart data', () => {
    const props: ProfileViewProps = {
      ...defaultProps,
      chartData: [
        { name: 'healthy', label: 'Healthy', variant: 'primary', value: 3 },
        { name: 'dead', label: 'Dead', variant: 'primary', value: 10 },
      ],
    };
    const { getByTestId } = renderWithTheme(
      <Router><ProfileView {...props} /></Router>,
    );

    expect(getByTestId('counters')).toHaveTextContent('03');
    expect(getByTestId('counters')).toHaveTextContent('10');
  });

  it('should display total amount of plants', () => {
    const props: ProfileViewProps = {
      ...defaultProps,
      counters: { total: 157 },
    };
    const { getByTestId } = renderWithTheme(
      <Router><ProfileView {...props} /></Router>,
    );

    expect(getByTestId('total_plants')).toHaveTextContent('157');
  });
});
