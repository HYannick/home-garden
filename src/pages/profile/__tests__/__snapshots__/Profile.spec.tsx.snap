// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page should render loading text properly 1`] = `
<div>
  <div>
    profile_page.loading
  </div>
</div>
`;

exports[`Profile Page should render properly 1`] = `
<div>
  <div>
    <div
      class="css-11xigr1"
    >
      <div
        class="css-10wldv8"
      >
        <svg
          viewBox="0 0 23.225 21.762"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.342 19.647L2.115 10.881m0 0l10.227-8.766m-8.617 9.032h18"
            fill="#4A4A4A"
            stroke="#4A4A4A"
            stroke-linecap="round"
            stroke-width="3"
          />
        </svg>
      </div>
      <div
        class="css-196st0l"
      >
        <span
          class="css-zrqali"
        >
          About you
        </span>
      </div>
      <div
        class="css-ov1ktg"
      />
    </div>
    <div
      class="css-1ue9cr"
    >
      <div
        class="css-1eyi4r2"
      >
        <div
          class="css-1b98xtx"
          data-testid="avatar"
        >
          <div
            class="css-vid29i"
            placeholder="#fff"
          >
            <img
              alt="avatar"
              class="css-10dr8mc"
              src="cute-kitten.jpg"
            />
          </div>
        </div>
        <div
          class="css-3xb32z"
          data-testid="username"
        >
          <h2>
            Mathilde
          </h2>
        </div>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-2w6ofm"
        >
          <div>
            <div
              class="recharts-wrapper"
              style="position: relative; cursor: default; width: 200px; height: 200px;"
            >
              <svg
                class="recharts-surface"
                height="200"
                version="1.1"
                viewBox="0 0 200 200"
                width="200"
              >
                <defs>
                  <clippath
                    id="recharts1-clip"
                  >
                    <rect
                      height="190"
                      width="190"
                      x="5"
                      y="5"
                    />
                  </clippath>
                </defs>
              </svg>
            </div>
            <div
              class="css-tuxdev"
              data-testid="total_plants"
            >
              <h4>
                undefined
              </h4>
              <span>
                Plants
              </span>
            </div>
          </div>
        </div>
        <div
          class="css-syznh7"
          data-testid="counters"
        />
        <a
          class="css-hmofew"
          href="/plants"
        >
          View plants
        </a>
      </div>
      <h4
        class="css-hrp93d"
      />
    </div>
  </div>
</div>
`;
